dbt/tests/data_tests/generic_tests/expect_date_frequency_ratio.sql

```diff
@@ -0,0 +1,52 @@
+{% test expect_date_frequency_ratio(
+    model,
+    compare_model,
+    multiplier,
+    date_column='report_date'
+) %}
+
+WITH model_years AS (
+    SELECT
+        EXTRACT(YEAR FROM {{ date_column }}) as year,
+        COUNT(DISTINCT {{ date_column }}) as date_count
+    FROM {{ model }}
+    WHERE {{ date_column }} IS NOT NULL
```

Comment:
Technically this `IS NOT NULL` check shouldn't be necessary, since `report_date` is part of the PK and can't be NULL. Maybe remove?